rust/src/lib/route.rs

```diff
@@ -104,6 +105,51 @@ impl Routes {
         }
         Ok(())
     }
+
+    pub(crate) fn resolve_next_hop_iface_ref(
+        &mut self,
+        merged_ifaces: &MergedInterfaces,
+    ) -> Result<(), NmstateError> {
+        let iface_name_search = InterfaceNameSearch::new(merged_ifaces);
+
+        if let Some(config_routes) = self.config.as_mut() {
+            for route in config_routes.iter_mut().filter(|r| !r.is_absent()) {
+                let route_display_str = route.to_string();
+                if let Some(nex_hop_iface) = route.next_hop_iface.as_mut() {
+                    let kernel_names = iface_name_search.get(nex_hop_iface);
+                    // Prefer kernel name as port name
+                    if kernel_names.contains(&nex_hop_iface.as_str()) {
+                        continue;
+                    }
+                    if kernel_names.len() > 1 {
+                        return Err(NmstateError::new(
+                            ErrorKind::InvalidArgument,
+                            format!(
+                                "Route '{}' defined with next \
+                                hop interface {} but \
+                                multiple interfaces are sharing the \
+                                this profile name",
+                                route_display_str,
+                                nex_hop_iface.as_str()
+                            ),
+                        ));
+                    } else if let Some(kernel_name) = kernel_names.first() {
+                        *nex_hop_iface = kernel_name.to_string();
+                    } else {
+                        return Err(NmstateError::new(
+                            ErrorKind::InvalidArgument,
+                            format!(
+                                "Route '{}': next hop interface {} no found",
```

Comment:
How about changing it to "Route '{}': next hop interface {} not found"?

Reply:
I don't see a difference between your suggested string and my code.

Reply:
~no~ not